Hazel/src/Platform/OpenGL/OpenGLRendererAPI.cpp

```diff
@@ -4,9 +4,34 @@
 #include <glad/glad.h>
 
 namespace Hazel {
+	
+#ifdef HZ_ENABLE_RENDERER_LOG
+	void OpenGLRendererAPI::OpenGLMessageCallback(
+		unsigned source,
+		unsigned type,
+		unsigned id,
+		unsigned severity,
+		int length,
+		const char* message,
+		const void* userParam)
+	{
+		switch (severity)
+		{
+			case GL_DEBUG_SEVERITY_HIGH:         HZ_CORE_CRITICAL(message); return;
+			case GL_DEBUG_SEVERITY_MEDIUM:       HZ_CORE_ERROR(message); return;
+			case GL_DEBUG_SEVERITY_LOW:          HZ_CORE_WARN(message); return;
+			case GL_DEBUG_SEVERITY_NOTIFICATION: HZ_CORE_TRACE(message); return;
```

Comment:
As of now, this is spitting out alot of unneeded detailed info. How about anther macro to define the level of debug, with a default to only output low up to high (and not notifications)?

For example:
```cpp
#ifdef HZ_ENABLE_RENDERER_LOG
	#ifndef HZ_ENABLE_RENDERER_LOG_LEVEL
		// default: warning and above
		#define HZ_ENABLE_RENDERER_LOG_LEVEL 2
	#endif

	void OpenGLRendererAPI::OpenGLMessageCallback(
		unsigned source,
		unsigned type,
		unsigned id,
		unsigned severity,
		int length,
		const char* message,
		const void* userParam)
	{
		switch (severity)
		{
			case GL_DEBUG_SEVERITY_HIGH:
				#if HZ_ENABLE_RENDERER_LOG_LEVEL <= 4
					HZ_CORE_CRITICAL(message);
				#endif
				return;
			case GL_DEBUG_SEVERITY_MEDIUM:
				#if HZ_ENABLE_RENDERER_LOG_LEVEL <= 3
					HZ_CORE_ERROR(message);
				#endif
				return;
			case GL_DEBUG_SEVERITY_LOW:
				#if HZ_ENABLE_RENDERER_LOG_LEVEL <= 2
					HZ_CORE_WARN(message);
				#endif
				return;
			case GL_DEBUG_SEVERITY_NOTIFICATION:
				#if HZ_ENABLE_RENDERER_LOG_LEVEL <= 1
					HZ_CORE_TRACE(message);
				#endif
				return;
		}
	}
#endif
```

Reply:
To ignore notifications, we can just use `glDebugMessageControl()`.  I'm confused by the macros tough.

Reply:
`glDebugMessageControl()` might indeed be the nicer solution, go for it. The macro's just enable/disable the logging of different severities... but `glDebugMessageControl()` is more readable.